corehq/apps/zapier/signals/receivers.py

```diff
@@ -22,13 +25,17 @@ def zapier_subscription_pre_save(sender, instance, *args, **kwargs):
             white_listed_form_xmlns=[instance.form_xmlns]
         )
 
-    elif instance.event_name == "new_case":
+    elif instance.event_name == consts.EventTypes.NEW_CASE:
         repeater = CaseRepeater(
             domain=instance.domain,
             url=instance.url,
             format='case_json',
             white_listed_case_types=[instance.case_type],
         )
+    else:
+        raise ImmediateHttpResponse(
+            HttpBadRequest('The passed event type is not valid.')
```

Comment:
👍 (nit: a bit of a nicer error message would have been to include a list of the event types that _are_ valid, but that's not important)